pkg/engine/internal/match.go

```diff
@@ -54,9 +54,15 @@ func checkResourceFilters(configuration config.Configuration, gvk schema.GroupVe
 func checkNamespacedPolicy(policy kyvernov1.PolicyInterface, resources ...unstructured.Unstructured) bool {
 	if policy.IsNamespaced() {
 		policyNamespace := policy.GetNamespace()
+		if policyNamespace == "" {
+			policyNamespace = "default"
+		}
 		for _, resource := range resources {
 			if resource.Object != nil {
 				resourceNamespace := resource.GetNamespace()
+				if resourceNamespace == "" {
+					resourceNamespace = "default"
+				}
```

Comment:
We can't consider the empty resource namespace as default because it can be a cluster-wide resource. The user must specify resource's namespace in the test file.

Reply:
Thank you for pointing that out! I didn't consider that an empty resource namespace could imply a cluster-wide resource. I appreciate the clarification.